fix: support for map image layers

cdoak-gh opened the pull request:
- add test for fields array
- removed extraneous 1=1 to beginning of where clause
- updated sample workflow to find map image service sublayers

cdoak-gh committed:
fix: support for map image layers
- add test for fields array
- removed extraneous 1=1 to beginning of where clause
- updated sample workflow to find map image service sublayers

rcooney commented on src/activities/GenerateQueryWhereClause.ts:
```diff
@@ -54,18 +57,22 @@ export default class GenerateWhereClause implements IActivityHandler {
         const definitionExpression = getDefinitionExpression(layer);
         if (definitionExpression) {
-            where = `${where} AND ${definitionExpression}`;
+            where = `${definitionExpression} AND ${where}`;
```
I think we need `(${definitionExpression}) AND (${where})` to avoid unintended effects of OR statements in either part.

rcooney commented on src/activities/GenerateQueryWhereClause.ts:
```diff
@@ -109,21 +116,23 @@ export default class GenerateWhereClause implements IActivityHandler {
 
-        return `${where}${fieldAndValue}`;
+        if(where.length > 0) {
+            return `${where} AND ${queryFields.field} ${queryFields.operator} ${value}`;
```
We should wrap each part in `( )` here too.

cdoak-gh committed:
fix:
GenerateQueryWhereClause
-isolate where clause statements with parentheses
-append AND only when approriate
utils:
- include 'sublayer' type for utility functions

rcooney commented on src/activities/GenerateQueryWhereClause.ts:
```diff
@@ -62,7 +62,12 @@ export default class GenerateWhereClause implements IActivityHandler {
         if (definitionExpression) {
-            where = `${definitionExpression} AND ${where}`;
+            if(where.length > 0) {
+                where = `${where} AND (${definitionExpression})`;
```
You need parentheses around both parts. Consider the case where:
- `where` is `Field1 = 1 OR Field2 = 2` 
- `definitionExpression` is `Field3 = 3`

This would currently result in
`Field1 = 1 OR Field2 = 2 AND (Field3 = 3)`
which (due to operator precedence) is equivalent to 
`Field1 = 1 OR (Field2 = 2 AND (Field3 = 3))`
which isn't what we want.

The same is required on line 135.

cdoak-gh committed:
fix: GenerateQueryWhereClause
- ensure operator precedence is maintained when building where clause

cdoak-gh committed:
Merge branch 'main' into field-collection-validation
